### PR TITLE
[5.1] Replaced the static uri parameter with a dynamic version

### PR DESCRIPTION
**Pull request by c4pone**
Cleaned up PR #10696, as told by @taylorotwell

### CODE REVIEW

File: src/Illuminate/Routing/ControllerInspector.php [hunk 1]
```diff
@@ -121,13 +121,27 @@ public function getPlainUri($name, $prefix)
...
+     * @param  ReflectionMethod  $method
+     * @param  string            $uri
      * @return string
      */
-    public function addUriWildcards($uri)
+    public function addUriParameters(ReflectionMethod $method, $uri)
     {
-        return $uri.'/{one?}/{two?}/{three?}/{four?}/{five?}';
+        foreach ($method->getParameters() as $parameter) {
+            //if the parameter has a class we assume it is
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ControllerInspector.php]: cs

File: src/Illuminate/Routing/ControllerInspector.php [hunk 2]
```diff
@@ -121,13 +121,27 @@ public function getPlainUri($name, $prefix)
     }
 
     /**
-     * Add wildcards to the given URI.
+     * Gets the parameters from a method in the controller.
      *
-     * @param  string  $uri
+     * @param  ReflectionMethod  $method
+     * @param  string            $uri
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ControllerInspector.php]: cs

File: src/Illuminate/Routing/ControllerInspector.php [hunk 3]
```diff
@@ -121,13 +121,27 @@ public function getPlainUri($name, $prefix)
     }
 
     /**
-     * Add wildcards to the given URI.
+     * Gets the parameters from a method in the controller.
      *
-     * @param  string  $uri
+     * @param  ReflectionMethod  $method
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ControllerInspector.php]: slash needed

**Comment by rkgrep** [comment]
I think, this should go to master

**Comment by darthmaim** [comment]
This breaks with parameters in the controller route, `Route::controller('foo/{id}')` with a controller containing `function bar($id)` generates `foo/{id}/bar/{id}`, should be `foo/{id}/bar`.

**Comment by rkgrep** [comment]
As @darthmaim mentioned, this will also break routes inside groups with parameters like
````php
Route::group(['prefix' => '{locale}'], function($router){ ... });
````

**Comment by GrahamCampbell** [comment]
Please send to the master then. :)

**Comment by GrahamCampbell** [comment]
Could you fix the cs issues at the same time too please?

**Comment by darthmaim** [comment]
`function test($foo='foo', $bar)` should probably also generate `test/{foo}/{bar}` instead of `test/{foo?}/{bar}`.

And model binding also doesn't work, `Router::model('foo', Foo::class)` + `function test(Foo $foo)` should generate `test/{foo}`, but `$foo` gets skipped because it has a typehint.

**Pull request closed without merging**